packages/marketplace/src/logic/Utils.ts

```diff
@@ -317,6 +318,109 @@ export const initColorShiftLoop = (schemes: SchemeIni) => {
   }, 60 * 1000);
 };
 
+export const getColorFromImage = async (image: HTMLImageElement, numColors : number) => {
+  const swatches = await Vibrant.from(image).maxColorCount(numColors).getPalette((err, palette) => {
+    if (err) {
+      console.error(err);
+      return;
+    }
+    return palette;
+  });
+  if (swatches.Vibrant) {
+    // remove the # from the hex
+    return swatches.Vibrant.hex.substring(1);
+
+  }
+  return "null";
+
+};
+
+export const generateColorPalette = async (mainColor : string, numColors : number) => {
+  // Generate a palette from https://www.thecolorapi.com/id?hex=0047AB&rgb=0,71,171&hsl=215,100%,34%&cmyk=100,58,0,33&format=html
+  const mode = getLocalStorageDataFromKey("marketplace:albumArtBasedColorsMode");
+  // Add a hyphen before any uppercase characters
+  const modeStr = mode.replace(/([A-Z])/g, "-$1").toLowerCase();
+  const palette = await Spicetify.CosmosAsync.get(`https://www.thecolorapi.com/scheme?hex=${mainColor}&mode=${modeStr}&count=${numColors}`);
```

Comment:
`CosmosAsync` includes a bunch of Spotify authorization stuff onto its header, couldn't this be done normally via `fetch`?

Reply:
Spotify got funky with it for whatever reason, it worked sometimes but not others and it didn't appear to be ratelimiting

Reply:
We used `fetch` everywhere else in Marketplace, not exclusively this. Surely it could not have been so.

Reply:
I could be misremembering with an error I experienced when attempting to use no api's, I don't think i am, but i will try tomorrow nontheless.